File: app/src/main/java/wangdaye/com/geometricweather/db/entities/HistoryEntity.java
package wangdaye.com.geometricweather.db.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import java.util.Date;

/**
 * History entity matching GreenDAO schema 62.
 */
@Entity(tableName = "HISTORY_ENTITY")
public class HistoryEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "ID")
    public Long id;

    @ColumnInfo(name = "CITY_ID")
    public String cityId;

    @ColumnInfo(name = "WEATHER_SOURCE")
    public String weatherSource;

    @ColumnInfo(name = "DATE")
    public Date date;

    @ColumnInfo(name = "TIME")
    public long time;

    @ColumnInfo(name = "DAYTIME_TEMPERATURE")
    public int daytimeTemperature;

    @ColumnInfo(name = "NIGHTTIME_TEMPERATURE")
    public int nighttimeTemperature;

    public HistoryEntity() {
    }
}

File: data/src/main/java/wangdaye/com/geometricweather/db/entities/HistoryEntity.kt
package wangdaye.com.geometricweather.db.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import kotlin.jvm.JvmField

/**
 * History entity matching GreenDAO schema 62.
 */
@Entity(tableName = "HISTORY_ENTITY")
class HistoryEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "ID")
    @JvmField
    var id: Long? = null

    @ColumnInfo(name = "CITY_ID")
    @JvmField
    var cityId: String? = null

    @ColumnInfo(name = "WEATHER_SOURCE")
    @JvmField
    var weatherSource: String? = null

    @ColumnInfo(name = "DATE")
    @JvmField
    var date: Date? = null

    @ColumnInfo(name = "TIME")
    @JvmField
    var time: Long = 0L

    @ColumnInfo(name = "DAYTIME_TEMPERATURE")
    @JvmField
    var daytimeTemperature: Int = 0

    @ColumnInfo(name = "NIGHTTIME_TEMPERATURE")
    @JvmField
    var nighttimeTemperature: Int = 0
}
